components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java

```diff
@@ -948,19 +952,28 @@ private static String getConsentRequiredScopesAsString(Set<String> consentRequir
     private static String getQueryString(OAuth2Parameters params, SessionDataCacheEntry entry) throws
             UnsupportedEncodingException, OAuthSystemException {
 
-        String queryString;
-        queryString = entry.getQueryString();
-        if (queryString.contains(REQUEST_URI) && params != null) {
+        StringBuilder queryStringBuilder = new StringBuilder();
+        queryStringBuilder.append(entry.getQueryString());
+        if (entry.getQueryString().contains(REQUEST_URI) && params != null) {
```

Comment:
is this null safe?

Reply:
This is getting called here [1] and it's already checking whether the entry is null. And this is the only usage of that method.

[1] - https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/57101951ae3f6b2aae4c480c4b6634bb07982044/components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java#L819-L820